libs/header/sync/sync.go

```diff
@@ -276,25 +276,25 @@ func (s *Syncer[H]) findHeaders(ctx context.Context, from, to uint64) ([]H, erro
 	if amount > s.Params.MaxRequestSize {
 		to, amount = from+s.Params.MaxRequestSize, s.Params.MaxRequestSize
 	}
-
 	out := make([]H, 0, amount)
-	for from < to {
+	for from <= to {
 		// if we have some range cached - use it
 		r, ok := s.pending.FirstRangeWithin(from, to)
 		if !ok {
-			hs, err := s.exchange.GetRangeByHeight(ctx, from, amount)
+			hs, err := s.exchange.GetRangeByHeight(ctx, from, to-from+1)
```

Comment:
Ok, so using the `amount` is a real bug here. Because it does not include changes to `from` from previous iterations